packages/knit/src/schema.test.ts

```diff
@@ -712,6 +712,49 @@ describe("client", () => {
         },
       });
     });
+
+    test("query for oneof with nested object", async () => {
+      const response = await client.fetch({
+        "spec.AllService": {
+          getAll: {
+            $: {},
+            oneof: {
+              oneofValue: oneof({
+                message: {
+                  id: {},
+                },
+                nestedMessage: {
+                  nested: {
+                    id: {},
+                  },
+                },
+              }),
+            },
+          },
+        },
+      });
+      type Actual = typeof response;
+      type Expected = {
+        "spec.AllService": {
+          getAll: {
+            oneof: {
+              oneofValue?: Oneof<{
+                message: {
+                  id: string;
+                };
+                nestedMessage: {
+                  nested?: {
+                    id: string;
+                  };
+                };
+              }>;
+            };
+          };
+        };
+      };
+      type Diff = DeepDiff<Actual, Expected>;
+      expectType<Equal<Diff, never>>(true);
```

Comment:
Ad it turns out, my previous in #301 was insufficient for the actual query which now still fails here. Back to the drawing board to figure out precisely why TS doesn't like this case.